src/Schema/Factories/DirectiveFactory.php

```diff
@@ -2,255 +2,37 @@
-            return $this->handler($directive->name->value);
-        })->filter(function ($handler) {
-            return $handler instanceof FieldResolver;
-        });
-
-        if ($resolvers->count() > 1) {
-            throw new DirectiveException(sprintf(
-                'Fields can only have 1 assigned resolver directive. %s has %s resolver directives [%s]',
-                data_get($field, 'name.value'),
-                $resolvers->count(),
-                collect($field->directives)->map(function (DirectiveNode $directive) {
-                    return $directive->name->value;
-                })->implode(', ')
-            ));
-        }
-
-        return $resolvers->first();
-    }
-
-    /**
-     * Check if field has a resolver directive.
-     *
-     * @param FieldDefinitionNode $field
-     *
-     * @return bool
-     */
-    public function hasFieldMiddleware($field)
-    {
-        return collect($field->directives)->map(function (DirectiveNode $directive) {
-            return $this->handler($directive->name->value);
-        })->reduce(function ($has, $handler) {
-            return $handler instanceof FieldMiddleware ? true : $has;
-        }, false);
-    }
-
-    /**
-     * Get middleware for field.
-     *
-     * @param FieldDefinitionNode $field
-     *
-     * @return \Illuminate\Support\Collection
-     */
-    public function fieldMiddleware($field)
-    {
-        return collect($field->directives)->map(function (DirectiveNode $directive) {
-            return $this->handler($directive->name->value);
-        })->filter(function ($handler) {
-            return $handler instanceof FieldMiddleware;
-        });
-    }
-
-    /**
-     * Get middleware for field arguments.
-     *
-     * @param InputValueDefinitionNode $arg
-     *
-     * @return \Illuminate\Support\Collection
-     */
-    public function argMiddleware(InputValueDefinitionNode $arg)
-    {
-        return collect($arg->directives)->map(function (DirectiveNode $directive) {
-            return $this->handler($directive->name->value);
-        })->filter(function ($handler) {
-            return $handler instanceof ArgMiddleware;
-        });
+     * Transform a directive definition into a directive.
+     *
+     * @param DirectiveDefinitionNode $directive
+     *
+     * @return Directive
+     */
+    public function toDirective(DirectiveDefinitionNode $directive)
+    {
+        return new Directive([
+            'name' => $directive->name->value,
+            'locations' => collect($directive->locations)->map(function ($location) {
+                return $location->value;
+            })->toArray(),
+            'args' => collect($directive->arguments)->map(function (InputValueDefinitionNode $argument) {
+                return new FieldArgument([
+                    'name' => $argument->name->value,
+                    'defaultValue' => $argument->defaultValue->value,
```

Comment:
Still testing, but it looks like this will fail if no default value is provided so we may want to wrap this with:


```php
'defaultValue' => data_get($argument, 'defaultValue.value', null),
```

Reply:
Good catch. Wrote a test - it failed - and fixed it!